posthog-android/src/test/java/com/posthog/android/PostHogFake.kt

```diff
@@ -76,17 +96,22 @@ public class PostHogFake : PostHogInterface {
         type: String,
         key: String,
         groupProperties: Map<String, Any>?,
+        distinctId: String?,
```

Comment:
`distinctId` is optional for Android since its cached in the disk/memory and required for Java, this has to be over documentation/method docs, otherwise I'd need to create another class and dupe a lot of code just because of this facility, is it worth it?

Reply:
"is it worth it"... 🤔 

it's a dev experience question I guess....

what does it mean for what I have to type?
what does it mean for how easy it is to make a mistake?
if i make a mistake how do I find out? will i get helpful errors

Reply:
If you call `capture` without `distinctId` on a Java backend app, it'll log in the console:

> PostHog.capture("event")
> vs
> PostHog.capture("event", distinctId = "123")

> "capture call not allowed, distinctId is invalid: $var."

I agree, I also prefer better dev UX but since both libs (core and Android) share the same interface, this is the issue we get hence my comment above.
If we prioritize better dev UX having an interface each lib we'd need to make major breaking changes, so we'd need to cut Android v4 as well, I was not aware or was an oversight from my side that `distinctId` has to be always passed during capture calls on Java since its stateless, this could be solved with scope management - thread local storage (if you set distinctId on thread/request X its always X, etc) but that's another big change not intended to be done now anyway.